### PR TITLE
Add support for JSON relational type

### PR DESCRIPTION
**Pull request by horbe**
## Summary

- [ x] Add support for `Json` relational data type

## How did you test this change?

- [ x] Test(s) added

Screen Shot 2023-04-24 at 8 56 36 AM
Screen Shot 2023-04-24 at 2 44 28 PM

### CODE REVIEW

**Review by akphi — CHANGES_REQUESTED** [review]
LGTM, just left a few minor comments

File: .changeset/plenty-crabs-try.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-application-studio': minor
+'@finos/legend-graph': minor
+---
+Add support for `Json` relational type.
```
[the remark this answers is not in the document]
Comment (horbe) [reply .changeset/plenty-crabs-try.md]: Done.

File: packages/legend-graph/src/graph-manager/__tests__/buildGraph/relational/SimpleRelationalGraphBuildSuccess.test.ts [hunk 2]
```diff
@@ -140,3 +141,10 @@ test(unitTest('SemiStructured relational type roundtrip'), async () => {
     pluginManager,
   );
 });
+
+test(unitTest('Json relational type roundtrip'), async () => {
```
[the remark this answers is not in the document]
Comment (horbe) [reply packages/legend-graph/src/graph-manager/__tests__/buildGraph/relational/SimpleRelationalGraphBuildSuccess.test.ts]: Done.

File: packages/legend-application-studio/src/components/editor/editor-group/connection-editor/DatabaseEditorHelper.tsx [hunk 3]
```diff
@@ -119,6 +122,7 @@ export const renderColumnTypeIcon = (
     return (
       <ClockIcon className="relation-source-tree__icon relation-source-tree__icon__time" />
     );
+  //     TODO HSO: Add icon?
```
[the remark this answers is not in the document]
Comment (horbe) [reply packages/legend-application-studio/src/components/editor/editor-group/connection-editor/DatabaseEditorHelper.tsx]: Got it.

**Review by akphi — APPROVED** [review]

**Pull request merged**